tofhir-server/src/main/scala/io/tofhir/server/service/SchemaDefinitionService.scala

```diff
@@ -124,4 +128,67 @@ class SchemaDefinitionService(schemaRepository: ISchemaRepository) extends LazyL
     JArray(rootElement +: elements)
   }
 
+  /**
+   * Infer the schema
```

Comment:
Let's explain the infer functionality here i.e. what it takes, what it does and what it returns at the end.